collectors/metrics/pkg/forwarder/forwarder.go

```diff
@@ -337,32 +395,34 @@ func (w *Worker) Reconfigure(cfg Config) error {
 	return nil
 }
 
+// TODO(saswatamcode): This is a relic of telemeter code, remove this.
+// There is no such utility to exposing this information as to what the last value of metrics sent was
+// Rarely would this be used as a tool for debugging, when you already have remote write metrics.
 func (w *Worker) LastMetrics() []*clientmodel.MetricFamily {
 	w.lock.Lock()
 	defer w.lock.Unlock()
 	return w.lastMetrics
 }
 
 func (w *Worker) Run(ctx context.Context) {
-	for {
-		// Ensure that the Worker does not access critical configuration during a reconfiguration.
-		w.lock.Lock()
-		wait := w.interval
-		// The critical section ends here.
-		w.lock.Unlock()
-
-		if err := w.forward(ctx); err != nil {
-			rlogger.Log(w.logger, rlogger.Error, "msg", "unable to forward results", "err", err)
-			wait = time.Minute
-		}
+	ticker := time.NewTicker(w.interval)
+	defer ticker.Stop()
 
+	for {
 		select {
 		// If the context is canceled, then we're done.
 		case <-ctx.Done():
+			ticker.Stop()
 			return
-		case <-time.After(wait):
+		case <-ticker.C:
+			if err := w.forward(ctx); err != nil {
```

Comment:
I think this should be exponential backoff retry not quite what this is.
Right now, if a request fails, regardless of interval you set, it will just wait a full minute and try again.